### PR TITLE
Ignore `described_class` being used as part of a constant

### LINKED ISSUE #799: error .. RSpec/DescribedClass cop was inspecting .. describe(CONSTANT)
Upgrading PaperTrail from 1.33.0 to 1.35.0, got this error:

```
An error occurred while RSpec/DescribedClass cop was 
inspecting .../spec/paper_trail/version_number_spec.rb:6:2.
```

In [`version_number_spec.rb`](https://github.com/paper-trail-gem/paper_trail/blob/da9c743831ff7b948680ee0ef847ad8516f1a206/spec/paper_trail/version_number_spec.rb) our `described_class` is a module named `VERSION`. 🤔 I admit, it's unusual (I didn't name it) but I think we can agree that rubocop should not crash :smile:

```
module PaperTrail
  ::RSpec.describe VERSION do
    describe "STRING" do
      it "joins the numbers into a period separated string" do
        expect(described_class::STRING).to eq(
          ...
```

Please see also [paper_trail/version_number.rb](https://github.com/paper-trail-gem/paper_trail/blob/da9c743831ff7b948680ee0ef847ad8516f1a206/lib/paper_trail/version_number.rb) if necessary.

I'm happy to work on this issue if someone can point me in the right direction. :heart:

### PR DESCRIPTION
**Pull request by pirj**
Sometimes, `described_class` is used as part of the constant, e.g. a namespace in the constant being analyzed, e.g.:

    expect(described_class::STRING).to eq('1.1.1')

In this case the constant was still detected as offensive, while it should not be.

Fixes #799

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by pirj** [comment]
@bquorning Please take a look.

File: lib/rubocop/cop/rspec/described_class.rb [hunk 1]
```diff
@@ -143,8 +145,7 @@ def full_const_name(node)
         #   collapse_namespace([:A, :B], [nil, :C)     # => [nil, :C]
         #   collapse_namespace([:A, :B], [nil, :B, :C) # => [nil, :B, :C]
         def collapse_namespace(namespace, const)
-          return const if namespace.empty?
-          return const if const.first.nil?
+          return const if namespace.empty? || const.first.nil?
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/described_class.rb]: I suspect this change is made _only_ to “please the cop”. I would rather increase the max file length than make unnecessary changes.

File: lib/rubocop/cop/rspec/described_class.rb [hunk 2]
```diff
@@ -102,11 +106,7 @@ def scope_change?(node)
         end
 
         def skippable_block?(node)
-          node.block_type? && !rspec_block?(node) && skip_blocks?
-        end
-
-        def skip_blocks?
-          cop_config['SkipBlocks']
+          node.block_type? && !rspec_block?(node) && cop_config['SkipBlocks']
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/described_class.rb]: This change also looks like a ClassLength “fix”, but it ends up making the code _less_ clear, in my opinion.

**Comment by pirj** [comment]
@bquorning Please take another look. I've introduced `.rubocop_todo.yml` with 104 lines for `Metrics/ClassLength`.

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# on 2019-08-06 21:29:02 +0300 using RuboCop version 0.70.0.
+# The point is for the user to remove these configuration records
+# one by one as the offenses are removed from the code base.
+# Note that changes in the inspected code, or installation of new
+# versions of RuboCop, may require this file to be generated again.
+
+# Offense count: 1
+# Configuration parameters: CountComments.
+Metrics/ClassLength:
+  Max: 104
```
Comment (bquorning) [inline .rubocop_todo.yml]: @Darhazer Is this ok with you?

**Pull request merged**